### PR TITLE
Cart UI and functionality

### PR DESCRIPTION
**Pull request by sharathnair9999**
Features included in the PR:

1. Cart UI
2. Cart Items
3. Cart count on navbar
4. Cart CRUD operations
5. Redirect to product from cart
6. Total Cart Price Summary

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]
Liked how you calculated all the results together into one reduce. Good work.

**Review by Kavita539 — APPROVED** [review]
- Code looks clean and structured.
- UI is also clean and minimal

File: src/components/Alert/Alert.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React from 'react'
 import "./Alert.css"
-const Alert = ({type, message}) => {
+const Alert = ({type, message, close}) => {
   return (
     <div className={`alert alert-${type}`}>
     <p className="alert-message">{message}</p>
-    <i className="far fa-times-circle alert-close fa-lg flex-and-center"></i>
+    <i className="far fa-times-circle alert-close fa-lg flex-and-center" onClick={close}></i>
```
Comment (ajay-mandaviya) [inline src/components/Alert/Alert.js:7]: avoid add Event-Listener directly on i tag

**Review by ajay-mandaviya — APPROVED** [review]
Everything looks good and code is more readable

File: src/components/Alert/Alert.js (hunk 1 above)
Comment (sharathnair9999) [reply src/components/Alert/Alert.js:7]: Yeah sure

**Pull request merged**